test/e2e/test_0123_import_scripts.py

```diff
@@ -0,0 +1,98 @@
+import filecmp
+import glob
+import os
+import re
+import shutil
+import unittest
+from zipfile import ZipFile
+
+import pytest
+
+from knora.dsplib.utils.onto_create_ontology import create_project
+from knora.dsplib.utils.xml_upload import xml_upload
+
+
+class TestImportScripts(unittest.TestCase):
+
+    def tearDown(self) -> None:
+        """
+        Remove generated data. This method is executed after every test method.
+        """
+        shutil.rmtree("docs/assets/0123-import-scripts-extracted", ignore_errors=True)
+        if os.path.isfile("docs/assets/0123-import-scripts/data-processed.xml"):
+            os.remove("docs/assets/0123-import-scripts/data-processed.xml")
+
+
+    def test_zip_folder(self) -> None:
+        """
+        A ZIP of the directory docs/assets/0123-import-scripts is available for download on docs.dasch.swiss. Make
+        sure that the ZIP's contents are identical to the directory
+        """
+        with ZipFile("docs/assets/0123-import-scripts.zip", "r") as zipfolder:
+            zipfolder.extractall("docs/assets/0123-import-scripts-extracted")
+        original_files = [x for x in glob.glob("docs/assets/0123-import-scripts/**", recursive=True) if os.path.isfile(x)]
+        extracted_files = [x for x in glob.glob("docs/assets/0123-import-scripts-extracted/0123-import-scripts/**", recursive=True)
+                           if os.path.isfile(x)]
+
+        # check that for every original file, there is exactly one extracted counterpart
+        original_files_expected = [re.sub(r"0123-import-scripts-extracted/", "", x) for x in extracted_files]
+        self.assertListEqual(sorted(original_files), sorted(original_files_expected), "docs/assets/0123-import-scripts.zip is outdated")
+
+        # check that the files have the same size and modification time
+        for orig, extr in zip(sorted(original_files), sorted(extracted_files)):
+            self.assertTrue(filecmp.cmp(orig, extr), "docs/assets/0123-import-scripts.zip is outdated")
+
+
+    @pytest.mark.filterwarnings("ignore")
+    def test_import_scripts(self) -> None:
+        """
+        Execute the import script in its directory, create the project on the DSP server, and upload the created XML to
+        the DSP server.
+        """
+        # execute the import script in its directory
+        old_working_directory = os.getcwd()
+        os.chdir("docs/assets/0123-import-scripts")
+        try:
+            with open("import-script.py") as f:
+                import_script = f.read()
+                exec(import_script, {})
```

Comment:
if you'd put all the code of the script in a `main()` method, you could simply import it and call `main()` here

Reply:
and you'd teach people the good practice of never to have top level scripts that aren't wrapped in a main method... ;-)

Reply:
Thanks for the hint, now I finally understand why there is a main() method in some Python scripts :-)